### ISSUE TITLE
[Typechecker | MaybeNonReproducable] Autocomplete will not find a named entity if the file has not yet been opened and changed in the current session using SQLLite autocomplete

### ISSUE THREAD

**Issue by lexidor** [facebook/hhvm#8569]
### HHVM Version
```
HipHop VM 4.24.0-dev (rel)
Compiler: 1568778476_831641667
Repo schema: 557dc2470ba5f41e75c58f5b1ea62ff53c729c14
hackc-2b00aad2b355c924e392ce3d004894215f1ad9ba-4.24.0-dev
OSS hack (no saved states)
```

### Operating System and Version
Ubuntu 18.04

### Standalone code, or other way to reproduce the problem

_Edit: A more reproducable exampled given here_: [example](https://github.com/facebook/hhvm/issues/8569#issuecomment-533714442)

```HACK
namespace Lex\Actor;

use namespace Lex\DB;

class User {
    // The exact classes are not important. This is merely how I found out about this.
    public static function fromDBRow(DB\IRo<<autocomplete_here> $row): void {
    }
}
```

#### Preconditions
These steps are not all required to have this problem occur, this is merely a step by step.

Have the above code without the incomplete function `fromDBRow` on disk.
```BASH
cd /path/to/project
hh_server --config symbolindex_search_provider=SqliteIndex .
```
Open vscode
From the vscode terminal: `hh_client 'No errors!'`
Reload vscode `ctrl+shift+p -> Developer: Reload Window`
Then type the incomplete function and attempt to autocomplete.

For generality, this *only* applies to namespaced named entities. I you make an class `\Foo`, it will be found without the need of loading the file and saving it.

#### What happens

 - I want this autocomplete to autocomplete to the named entity `\Lex\DB\IRow`.
 - If I have not yet opened the file in which IRow is defined in my currrent session and forced the typechecker to look at it by saving the file, IRow (and any other named entities that live in that namespace) don't exist in autocomplete.
 - If I open the IRow file and save it (closing the IRow file is now optional) and go back to the User file, I can autocomplete again. That now only shows `IRow` not any other classes and interfaces.

It is however important to note that the typechecker does know about IRow. If I type out the name myself, there is no Unbound Name error.

This did not happen a few days ago. Everything worked as intended. It might be something that really changed or just something that doesn't happen every time. Even if a reboot fixes the issue, I'll leave this github issue up, in case someone else has the same issue.

### Expected result
I expect the typechecker to autocomplete to all named entities it knows about.

I hope that this means something to `Ted Spence`. I am sadly unable to `@` autocomplete to your github account, so I can't ping you. I'll reference this issue to you via different means.

### Debugging information

This project is very empty for a reason. I made this with the express purpose of just toying around with the new features last week.

#### File structure
```
$ tree src
src
├── code
│   ├── Actor
│   │   ├── IUser.hack
│   │   └── User.hack
│   └── DB
│       ├── _Connection
│       │   ├── Connection.hack
│       │   └── IConnection.hack
│       ├── _Connector
│       │   ├── Connector.hack
│       │   └── IConnector.hack
│       ├── _Exception
│       │   └── Exception.hack
│       ├── _Result
│       │   ├── IResult.hack
│       │   └── Result.hack
│       └── _Row
│           ├── IRow.hack
│           └── Row.hack
├── lib
│   ├── Cast
│   │   └── mix.hack
│   └── Iter
│       └── for_each.hack
└── phq
    └── typed_echo.hack
```

#### .hhconfig contents
`<<empty file>>`

#### Other dependencies
```
cat composer.lock 
{
    "_readme": [
        "This file locks the dependencies of your project to a known state",
        "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#installing-dependencies",
        "This file is @generated automatically"
    ],
    "content-hash": "92afaefdb98bf77fce97f55944e0ee48",
    "packages": [
        {
            "name": "hhvm/hhvm-autoload",
            "version": "v2.0.8",
            "source": {
                "type": "git",
                "url": "https://github.com/hhvm/hhvm-autoload.git",
                "reference": "b5c0fb858e554c9b36a349aab2775c3914c7b2bf"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/hhvm/hhvm-autoload/zipball/b5c0fb858e554c9b36a349aab2775c3914c7b2bf",
                "reference": "b5c0fb858e554c9b36a349aab2775c3914c7b2bf",
                "shasum": ""
            },
            "require": {
                "composer-plugin-api": "^1.0",
                "hhvm": "^4.5",
                "hhvm/hsl": "^4.0"
            },
            "replace": {
                "facebook/hhvm-autoload": "1.*"
            },
            "require-dev": {
                "facebook/fbexpect": "^2.1",
                "hhvm/hacktest": "^1.0"
            },
            "bin": [
                "bin/hh-autoload",
                "bin/hh-autoload.hack"
            ],
            "type": "composer-plugin",
            "extra": {
                "class": [
                    "Facebook\\AutoloadMap\\ComposerPlugin"
                ],
                "branch-alias": {
                    "dev-master": "2.x-dev"
                }
            },
            "autoload": {
                "classmap": [
                    "src/ComposerPlugin.php"
                ]
            },
            "notification-url": "https://packagist.org/downloads/",
            "time": "2019-08-28T15:08:11+00:00"
        },
        {
            "name": "hhvm/hsl",
            "version": "v4.15.2",
            "source": {
                "type": "git",
                "url": "https://github.com/hhvm/hsl.git",
                "reference": "669c0c4297bcfdff0f2af9a86ed730511fbc0a96"
            },
            "dist": {
                "type": "zip",
                "url": "https://api.github.com/repos/hhvm/hsl/zipball/669c0c4297bcfdff0f2af9a86ed730511fbc0a96",
                "reference": "669c0c4297bcfdff0f2af9a86ed730511fbc0a96",
                "shasum": ""
            },
            "require": {
                "hhvm": "^4.18"
            },
            "require-dev": {
                "facebook/fbexpect": "^2.5.1",
                "hhvm/hacktest": "^1.0",
                "hhvm/hhvm-autoload": "^2.0"
            },
            "type": "library",
            "extra": {
                "branch-alias": {
                    "dev-master": "4.x-dev"
                }
            },
            "notification-url": "https://packagist.org/downloads/",
            "license": [
                "MIT"
            ],
            "description": "The Hack Standard Library",
            "time": "2019-08-27T21:27:21+00:00"
        }
    ],
    "packages-dev": [],
    "aliases": [],
    "minimum-stability": "stable",
    "stability-flags": [],
    "prefer-stable": false,
    "prefer-lowest": false,
    "platform": [],
    "platform-dev": []
}
```

#### hh_server output

```
<<<<<BEFORE STARTING VSCODE>>>>>

[2019-09-20 22:57:00.454] Server version is not set
[2019-09-20 22:57:00.454] Loading config exception: (Sys_error "/etc/hh.conf: No such file or directory")
[2019-09-20 22:57:00.454] Could not load config at /etc/hh.conf
Config overrides:
symbolindex_search_provider = SqliteIndex

The combined config:
symbolindex_search_provider = SqliteIndex
Informant using dummy - resigning
[2019-09-20 22:57:00.460] Starting first server
[2019-09-20 22:57:00.460] About to spawn typechecker daemon. Logs will go here.
[2019-09-20 22:57:00.474] Just started typechecker server with pid: 10255.
[2019-09-20 22:57:00.561] Server version is not set
[2019-09-20 22:57:00.561] Loading config exception: (Sys_error "/etc/hh.conf: No such file or directory")
[2019-09-20 22:57:00.561] Could not load config at /etc/hh.conf
Config overrides:
symbolindex_search_provider = SqliteIndex

The combined config:
symbolindex_search_provider = SqliteIndex
[2019-09-20 22:57:00.562] Version: hackc-2b00aad2b355c924e392ce3d004894215f1ad9ba-4.24.0-dev
[2019-09-20 22:57:00.562] Hostname: lexidor
[2019-09-20 22:57:00.564] Using dfind
[2019-09-20 22:57:00.565] Initializing Server (This might take some time)
[2019-09-20 22:57:00.565] Init id: ppXHyjPIU6
[2019-09-20 22:57:00.565] Initing with approach: Local_config_saved_state_disabled
[2019-09-20 22:57:00.565] indexing
[2019-09-20 22:57:00.565] indexing: 0.000003
[2019-09-20 22:57:00.565] parsing
[2019-09-20 22:57:01.107] Heap size: 1587904
[2019-09-20 22:57:01.108] Parsing: 0.542309
[2019-09-20 22:57:01.108] Updating file dependencies...
[2019-09-20 22:57:01.111] Updated file dependencies: 0.003142
[2019-09-20 22:57:01.111] resolving symbol references
[2019-09-20 22:57:01.136] Heap size: 2448512
[2019-09-20 22:57:01.136] Naming: 0.025373
[2019-09-20 22:57:01.138] Unable to fetch sqlite symbol index: Saved states are not supported in this build.
[2019-09-20 22:57:01.138] Generating [/tmp/symbolindex.zSoptzSlamppzShtdocszSphq.1569013021.db] from repository [/opt/lampp/htdocs/phq/]
[2019-09-20 22:57:01.142] Scanned repository folder [/opt/lampp/htdocs/phq/] in  [0.0 secs]
[2019-09-20 22:57:01.145] Scanned HHI folder [/tmp/hh_server/hhi_19470cda] in  [0.0 secs]
[2019-09-20 22:57:01.213] Parsed 206 files in  [0.1 secs]
[2019-09-20 22:57:01.224] Wrote 5133 symbols to sqlite in  [0.0 secs]
[2019-09-20 22:57:01.224] Initialized symbol index sqlite: [/tmp/symbolindex.zSoptzSlamppzShtdocszSphq.1569013021.db]
[2019-09-20 22:57:01.224] Search provider set to [Sqlite] based on configuration value [SqliteIndex]
[2019-09-20 22:57:01.224] Updated search index for symbols in 248 files:: 0.000013
[2019-09-20 22:57:01.224] Done updating search index
[2019-09-20 22:57:01.225] Dependency table load factor: 0 / 131072 (0.00)
[2019-09-20 22:57:01.225] Hashtable load factor: 8933 / 262144 (0.03) with 8933 nonempty slots
[2019-09-20 22:57:01.225] Waiting for daemon(s) to be ready...
[2019-09-20 22:57:01.225] Server is partially ready
[2019-09-20 22:57:01.225] Took 0.660090 seconds.
******************************************
[2019-09-20 22:57:01.326] Check kind: Full_check
[2019-09-20 22:57:01.326] Processing changes to 0 files
[2019-09-20 22:57:01.326] Processing deferred type decl for 0 files
[2019-09-20 22:57:01.326] Processing deferred typechecking for 248 files
[2019-09-20 22:57:01.326] Begin Parsing 0 files
[2019-09-20 22:57:01.326] Parsing 0 files: 0.000169
[2019-09-20 22:57:01.326] Heap size: 2448512
[2019-09-20 22:57:01.326] Begin Updating deps
[2019-09-20 22:57:01.326] Updating deps: 0.000080
[2019-09-20 22:57:01.326] Begin Naming
[2019-09-20 22:57:01.326] Naming: 0.000070
[2019-09-20 22:57:01.326] Begin Type declaration (phase 1) for 0 files
[2019-09-20 22:57:01.326] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 22:57:01.329] Type declaration (phase 1) for 0 files: 0.002245
[2019-09-20 22:57:01.329] Heap size: 2448512
[2019-09-20 22:57:01.329] Begin Type declaration (phase 2) for 0 files
[2019-09-20 22:57:01.329] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 22:57:01.329] Invalidating (but not recomputing) declarations in 0 files
[2019-09-20 22:57:01.330] Type declaration (phase 2) for 0 files: 0.001597
[2019-09-20 22:57:01.330] Heap size: 2448512
[2019-09-20 22:57:01.330] Begin evaluating prechecked changes
[2019-09-20 22:57:01.330] Evaluating prechecked changes: 0.000066
[2019-09-20 22:57:01.330] Extending the FAST by 248 additional files
[2019-09-20 22:57:01.330] Extending FAST sequentially (in memory)...
[2019-09-20 22:57:01.334] Extended FAST: 0.003684
[2019-09-20 22:57:01.334] Begin Typechecking 248 files
[2019-09-20 22:57:01.334] Calculated whether should do remote type checking (false): 0.000005
[2019-09-20 22:57:01.334] Type checking service will process files in parallel
[2019-09-20 22:57:01.659] Typechecked 248 files: 0.328310
[2019-09-20 22:57:01.659] Total: 0.332537

[2019-09-20 22:57:01.659] Dependency table load factor: 5437 / 131072 (0.04)
[2019-09-20 22:57:01.659] Hashtable load factor: 17503 / 262144 (0.07) with 17503 nonempty slots
[2019-09-20 22:57:01.659] Heap size: 4327552
[2019-09-20 22:57:01.659] Server is READY
[2019-09-20 22:57:01.659] Took 1.093898 seconds to initialize.
[2019-09-20 22:57:01.659] Recheck id: kAmvQ9r5n7
[2019-09-20 22:57:01.659] Finished recheck_loop; no diag

<<<<<STARTING VSCODE AND RUNNING HH_CLIENT>>>>>

[2019-09-20 22:57:56.319] Got priority request for typechecker. Prior request 56.0 seconds ago
[2019-09-20 22:57:56.321] Diag_subscribe: SUBSCRIBE 0
[2019-09-20 22:57:56.327] Diag_subscribe: open, remains active
[2019-09-20 22:58:02.786] Got default request for typechecker. Prior request 6.0 seconds ago


<<<<<TYPING DB\ AND REQUESTING AUTOCOMPLETE>>>>>
[2019-09-20 22:59:02.615] Check kind: Lazy_check
[2019-09-20 22:59:02.615] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 22:59:02.615] Begin Parsing 1 files
[2019-09-20 22:59:02.616] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 22:59:02.617] Updated search index for symbols in 1 files:: 0.000872
[2019-09-20 22:59:02.618] Done updating search index
[2019-09-20 22:59:02.618] Parsing 1 files: 0.003047
[2019-09-20 22:59:02.618] Heap size: 4328576
[2019-09-20 22:59:02.618] Begin Updating deps
[2019-09-20 22:59:02.618] Updating deps: 0.000078
[2019-09-20 22:59:02.618] Begin Naming
[2019-09-20 22:59:02.618] Naming: 0.000235
[2019-09-20 22:59:02.618] Begin Type declaration (phase 1) for 0 files
[2019-09-20 22:59:02.618] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 22:59:02.620] Type declaration (phase 1) for 0 files: 0.001715
[2019-09-20 22:59:02.620] Heap size: 4328576
[2019-09-20 22:59:02.620] Begin Type declaration (phase 2) for 0 files
[2019-09-20 22:59:02.620] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 22:59:02.620] Invalidating (but not recomputing) declarations in 0 files
[2019-09-20 22:59:02.622] Type declaration (phase 2) for 0 files: 0.001604
[2019-09-20 22:59:02.622] Heap size: 4328576
[2019-09-20 22:59:02.622] Begin evaluating prechecked changes
[2019-09-20 22:59:02.622] Evaluating prechecked changes: 0.000062
[2019-09-20 22:59:02.622] Begin Typechecking 0 files
[2019-09-20 22:59:02.622] Calculated whether should do remote type checking (false): 0.000004
[2019-09-20 22:59:02.622] Type checking service will process files sequentially
[2019-09-20 22:59:02.623] Typechecked 0 files: 0.001691
[2019-09-20 22:59:02.623] Total: 0.008432

[2019-09-20 22:59:02.624] Finished recheck_loop; sending push message
******************************************
[2019-09-20 22:59:03.731] Check kind: Full_check
[2019-09-20 22:59:03.731] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 22:59:03.731] Processing deferred type decl for 0 files
[2019-09-20 22:59:03.731] Processing deferred typechecking for 0 files
[2019-09-20 22:59:03.731] Begin Parsing 1 files
[2019-09-20 22:59:03.733] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 22:59:03.733] Updated search index for symbols in 1 files:: 0.000438
[2019-09-20 22:59:03.733] Done updating search index
[2019-09-20 22:59:03.733] Parsing 1 files: 0.002142
[2019-09-20 22:59:03.733] Heap size: 4329408
[2019-09-20 22:59:03.733] Begin Updating deps
[2019-09-20 22:59:03.733] Updating deps: 0.000132
[2019-09-20 22:59:03.734] Begin Naming
[2019-09-20 22:59:03.734] Naming: 0.000220
[2019-09-20 22:59:03.734] Begin Type declaration (phase 1) for 1 files
[2019-09-20 22:59:03.734] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 22:59:03.736] Type declaration (phase 1) for 1 files: 0.002153
[2019-09-20 22:59:03.736] Heap size: 4330496
[2019-09-20 22:59:03.736] Begin Type declaration (phase 2) for 1 files
[2019-09-20 22:59:03.736] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 22:59:03.736] Invalidating (but not recomputing) declarations in 0 files
[2019-09-20 22:59:03.738] Type declaration (phase 2) for 1 files: 0.002598
[2019-09-20 22:59:03.738] Heap size: 4331392
[2019-09-20 22:59:03.738] Begin evaluating prechecked changes
[2019-09-20 22:59:03.738] Evaluating prechecked changes: 0.000098
[2019-09-20 22:59:03.739] Begin Typechecking 1 files
[2019-09-20 22:59:03.739] Calculated whether should do remote type checking (false): 0.000004
[2019-09-20 22:59:03.739] Type checking service will process files sequentially
[2019-09-20 22:59:03.741] Typechecked 1 files: 0.002810
[2019-09-20 22:59:03.741] Total: 0.010153

[2019-09-20 22:59:03.742] Recheck id: NLoMz2Yh65
[2019-09-20 22:59:03.742] Finished recheck_loop; sending push message
******************************************
[2019-09-20 22:59:04.289] Check kind: Lazy_check
[2019-09-20 22:59:04.289] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 22:59:04.289] Begin Parsing 1 files
[2019-09-20 22:59:04.290] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 22:59:04.290] Updated search index for symbols in 1 files:: 0.000441
[2019-09-20 22:59:04.290] Done updating search index
[2019-09-20 22:59:04.291] Parsing 1 files: 0.001799
[2019-09-20 22:59:04.291] Heap size: 4331904
[2019-09-20 22:59:04.291] Begin Updating deps
[2019-09-20 22:59:04.291] Updating deps: 0.000066
[2019-09-20 22:59:04.291] Begin Naming
[2019-09-20 22:59:04.291] Naming: 0.000088
[2019-09-20 22:59:04.291] Begin Type declaration (phase 1) for 0 files
[2019-09-20 22:59:04.291] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 22:59:04.293] Type declaration (phase 1) for 0 files: 0.001797
[2019-09-20 22:59:04.293] Heap size: 4331904
[2019-09-20 22:59:04.293] Begin Type declaration (phase 2) for 0 files
[2019-09-20 22:59:04.293] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 22:59:04.293] Invalidating (but not recomputing) declarations in 0 files
[2019-09-20 22:59:04.295] Type declaration (phase 2) for 0 files: 0.002034
[2019-09-20 22:59:04.295] Heap size: 4331904
[2019-09-20 22:59:04.295] Begin evaluating prechecked changes
[2019-09-20 22:59:04.295] Evaluating prechecked changes: 0.000097
[2019-09-20 22:59:04.295] Extending the FAST by 1 additional files
[2019-09-20 22:59:04.295] Extending FAST sequentially (in memory)...
[2019-09-20 22:59:04.295] Extended FAST: 0.000026
[2019-09-20 22:59:04.295] Begin Typechecking 0 files
[2019-09-20 22:59:04.295] Calculated whether should do remote type checking (false): 0.000004
[2019-09-20 22:59:04.295] Type checking service will process files sequentially
[2019-09-20 22:59:04.297] Typechecked 0 files: 0.002161
[2019-09-20 22:59:04.297] Total: 0.008042

[2019-09-20 22:59:04.297] Finished recheck_loop; sending push message
******************************************
[2019-09-20 22:59:05.394] Check kind: Full_check
[2019-09-20 22:59:05.394] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 22:59:05.394] Processing deferred type decl for 0 files
[2019-09-20 22:59:05.394] Processing deferred typechecking for 0 files
[2019-09-20 22:59:05.394] Begin Parsing 1 files
[2019-09-20 22:59:05.395] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 22:59:05.396] Updated search index for symbols in 1 files:: 0.000469
[2019-09-20 22:59:05.396] Done updating search index
[2019-09-20 22:59:05.396] Parsing 1 files: 0.001610
[2019-09-20 22:59:05.396] Heap size: 4332736
[2019-09-20 22:59:05.396] Begin Updating deps
[2019-09-20 22:59:05.396] Updating deps: 0.000077
[2019-09-20 22:59:05.396] Begin Naming
[2019-09-20 22:59:05.396] Naming: 0.000157
[2019-09-20 22:59:05.396] Begin Type declaration (phase 1) for 1 files
[2019-09-20 22:59:05.396] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 22:59:05.399] Type declaration (phase 1) for 1 files: 0.002777
[2019-09-20 22:59:05.399] Heap size: 4333824
[2019-09-20 22:59:05.399] Begin Type declaration (phase 2) for 1 files
[2019-09-20 22:59:05.399] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 22:59:05.399] Invalidating (but not recomputing) declarations in 0 files
[2019-09-20 22:59:05.402] Type declaration (phase 2) for 1 files: 0.003199
[2019-09-20 22:59:05.402] Heap size: 4334720
[2019-09-20 22:59:05.402] Begin evaluating prechecked changes
[2019-09-20 22:59:05.402] Evaluating prechecked changes: 0.000095
[2019-09-20 22:59:05.402] Begin Typechecking 1 files
[2019-09-20 22:59:05.402] Calculated whether should do remote type checking (false): 0.000004
[2019-09-20 22:59:05.402] Type checking service will process files sequentially
[2019-09-20 22:59:05.405] Typechecked 1 files: 0.002943
[2019-09-20 22:59:05.405] Total: 0.010858

[2019-09-20 22:59:05.406] Recheck id: 4IXhkUeAD0
[2019-09-20 22:59:05.406] Finished recheck_loop; sending push message

<<<<<OPENING UP THE IROW FILE>>>>>

[2019-09-20 23:01:45.834] Diag_subscribe: open, remains active
******************************************
[2019-09-20 23:01:52.327] Check kind: Full_check
[2019-09-20 23:01:52.327] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/DB/_Row/IRow.hack
[2019-09-20 23:01:52.327] Processing deferred type decl for 0 files
[2019-09-20 23:01:52.327] Processing deferred typechecking for 0 files
[2019-09-20 23:01:52.327] Begin Parsing 1 files
[2019-09-20 23:01:52.331] src/code/DB/_Row/IRow.hack - classes: \Lex\DB\IRow
[2019-09-20 23:01:52.332] Updated search index for symbols in 1 files:: 0.000801
[2019-09-20 23:01:52.332] Done updating search index
[2019-09-20 23:01:52.332] Parsing 1 files: 0.005190
[2019-09-20 23:01:52.332] Heap size: 4336640
[2019-09-20 23:01:52.332] Begin Updating deps
[2019-09-20 23:01:52.332] Updating deps: 0.000073
[2019-09-20 23:01:52.332] Begin Naming
[2019-09-20 23:01:52.332] Naming: 0.000190
[2019-09-20 23:01:52.332] Begin Type declaration (phase 1) for 1 files
[2019-09-20 23:01:52.332] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 23:01:52.335] Type declaration (phase 1) for 1 files: 0.002529
[2019-09-20 23:01:52.335] Heap size: 4339456
[2019-09-20 23:01:52.335] Begin Type declaration (phase 2) for 1 files
[2019-09-20 23:01:52.335] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 23:01:52.335] Invalidating (but not recomputing) declarations in 0 files
[2019-09-20 23:01:52.337] Type declaration (phase 2) for 1 files: 0.002485
[2019-09-20 23:01:52.337] Heap size: 4342080
[2019-09-20 23:01:52.337] Begin evaluating prechecked changes
[2019-09-20 23:01:52.337] Evaluating prechecked changes: 0.000090
[2019-09-20 23:01:52.337] Begin Typechecking 1 files
[2019-09-20 23:01:52.337] Calculated whether should do remote type checking (false): 0.000004
[2019-09-20 23:01:52.337] Type checking service will process files sequentially
[2019-09-20 23:01:52.340] Typechecked 1 files: 0.003105
[2019-09-20 23:01:52.341] Total: 0.013662

[2019-09-20 23:01:52.341] Recheck id: fyjWdMrIby
[2019-09-20 23:01:52.341] Finished recheck_loop; is_empty errors

<<<<<SUCCESSFULLY AUTOCOMPLETING TO DB\IROW AND TYPING $row>>>>>
******************************************
[2019-09-20 23:02:37.869] Check kind: Lazy_check
[2019-09-20 23:02:37.869] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 23:02:37.869] Begin Parsing 1 files
[2019-09-20 23:02:37.870] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 23:02:37.871] Updated search index for symbols in 1 files:: 0.000418
[2019-09-20 23:02:37.871] Done updating search index
[2019-09-20 23:02:37.871] Parsing 1 files: 0.001766
[2019-09-20 23:02:37.871] Heap size: 4342656
[2019-09-20 23:02:37.871] Begin Updating deps
[2019-09-20 23:02:37.871] Updating deps: 0.000068
[2019-09-20 23:02:37.871] Begin Naming
[2019-09-20 23:02:37.871] Naming: 0.000088
[2019-09-20 23:02:37.871] Begin Type declaration (phase 1) for 0 files
[2019-09-20 23:02:37.871] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 23:02:37.873] Type declaration (phase 1) for 0 files: 0.001873
[2019-09-20 23:02:37.873] Heap size: 4342656
[2019-09-20 23:02:37.873] Begin Type declaration (phase 2) for 0 files
[2019-09-20 23:02:37.873] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 23:02:37.873] Invalidating (but not recomputing) declarations in 0 files
[2019-09-20 23:02:37.875] Type declaration (phase 2) for 0 files: 0.001953
[2019-09-20 23:02:37.875] Heap size: 4342656
[2019-09-20 23:02:37.875] Begin evaluating prechecked changes
[2019-09-20 23:02:37.875] Evaluating prechecked changes: 0.000084
[2019-09-20 23:02:37.875] Extending the FAST by 1 additional files
[2019-09-20 23:02:37.875] Extending FAST sequentially (in memory)...
[2019-09-20 23:02:37.875] Extended FAST: 0.000031
[2019-09-20 23:02:37.875] Begin Typechecking 0 files
[2019-09-20 23:02:37.875] Calculated whether should do remote type checking (false): 0.000005
[2019-09-20 23:02:37.875] Type checking service will process files sequentially
[2019-09-20 23:02:37.877] Typechecked 0 files: 0.002231
[2019-09-20 23:02:37.877] Total: 0.008063

[2019-09-20 23:02:37.877] Finished recheck_loop; sending push message
******************************************
[2019-09-20 23:02:39.103] Check kind: Full_check
[2019-09-20 23:02:39.103] Processing changes to 2 files
[2019-09-20 23:02:39.103] Processing deferred type decl for 0 files
[2019-09-20 23:02:39.103] Processing deferred typechecking for 0 files
[2019-09-20 23:02:39.103] Begin Parsing 2 files
[2019-09-20 23:02:39.128] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 23:02:39.128] src/code/DB/_Row/IRow.hack - classes: \Lex\DB\IRow
[2019-09-20 23:02:39.129] Updated search index for symbols in 2 files:: 0.001092
[2019-09-20 23:02:39.129] Done updating search index
[2019-09-20 23:02:39.129] Parsing 2 files: 0.026229
[2019-09-20 23:02:39.129] Heap size: 4345088
[2019-09-20 23:02:39.129] Begin Updating deps
[2019-09-20 23:02:39.129] Updating deps: 0.000129
[2019-09-20 23:02:39.129] Begin Naming
[2019-09-20 23:02:39.130] Naming: 0.000203
[2019-09-20 23:02:39.130] Begin Type declaration (phase 1) for 2 files
[2019-09-20 23:02:39.130] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 23:02:39.132] Type declaration (phase 1) for 2 files: 0.002819
[2019-09-20 23:02:39.132] Heap size: 4348992
[2019-09-20 23:02:39.132] Begin Type declaration (phase 2) for 2 files
[2019-09-20 23:02:39.133] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 23:02:39.133] Invalidating (but not recomputing) declarations in 0 files
[2019-09-20 23:02:39.135] Type declaration (phase 2) for 2 files: 0.002723
[2019-09-20 23:02:39.135] Heap size: 4352512
[2019-09-20 23:02:39.135] Begin evaluating prechecked changes
[2019-09-20 23:02:39.135] Evaluating prechecked changes: 0.000092
[2019-09-20 23:02:39.135] Begin Typechecking 2 files
[2019-09-20 23:02:39.135] Calculated whether should do remote type checking (false): 0.000004
[2019-09-20 23:02:39.135] Type checking service will process files sequentially
[2019-09-20 23:02:39.139] Typechecked 2 files: 0.004155
[2019-09-20 23:02:39.139] Total: 0.036350

[2019-09-20 23:02:39.140] Recheck id: bQP32DDJss
[2019-09-20 23:02:39.140] Finished recheck_loop; is_empty errors
******************************************
[2019-09-20 23:02:40.984] Check kind: Lazy_check
[2019-09-20 23:02:40.984] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 23:02:40.984] Begin Parsing 1 files
[2019-09-20 23:02:40.985] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 23:02:40.988] Updated search index for symbols in 1 files:: 0.002813
[2019-09-20 23:02:40.989] Done updating search index
[2019-09-20 23:02:40.989] Parsing 1 files: 0.005032
[2019-09-20 23:02:40.989] Heap size: 4352768
[2019-09-20 23:02:40.989] Begin Updating deps
[2019-09-20 23:02:40.989] Updating deps: 0.000085
[2019-09-20 23:02:40.989] Begin Naming
[2019-09-20 23:02:40.989] Naming: 0.000103
[2019-09-20 23:02:40.989] Begin Type declaration (phase 1) for 0 files
[2019-09-20 23:02:40.989] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 23:02:40.991] Type declaration (phase 1) for 0 files: 0.001749
[2019-09-20 23:02:40.991] Heap size: 4352768
[2019-09-20 23:02:40.991] Begin Type declaration (phase 2) for 0 files
[2019-09-20 23:02:40.991] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 23:02:40.991] Invalidating (but not recomputing) declarations in 0 files
[2019-09-20 23:02:40.993] Type declaration (phase 2) for 0 files: 0.001799
[2019-09-20 23:02:40.993] Heap size: 4352768
[2019-09-20 23:02:40.993] Begin evaluating prechecked changes
[2019-09-20 23:02:40.993] Evaluating prechecked changes: 0.000066
[2019-09-20 23:02:40.993] Begin Typechecking 0 files
[2019-09-20 23:02:40.993] Calculated whether should do remote type checking (false): 0.000004
[2019-09-20 23:02:40.993] Type checking service will process files sequentially
[2019-09-20 23:02:40.995] Typechecked 0 files: 0.001842
[2019-09-20 23:02:40.995] Total: 0.010676

[2019-09-20 23:02:40.995] Finished recheck_loop; sending push message
******************************************
[2019-09-20 23:02:41.729] Check kind: Lazy_check
[2019-09-20 23:02:41.729] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 23:02:41.729] Begin Parsing 1 files
[2019-09-20 23:02:41.730] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 23:02:41.731] Updated search index for symbols in 1 files:: 0.000433
[2019-09-20 23:02:41.731] Done updating search index
[2019-09-20 23:02:41.731] Parsing 1 files: 0.002105
[2019-09-20 23:02:41.731] Heap size: 4354624
[2019-09-20 23:02:41.731] Begin Updating deps
[2019-09-20 23:02:41.731] Updating deps: 0.000084
[2019-09-20 23:02:41.731] Begin Naming
[2019-09-20 23:02:41.731] Naming: 0.000144
[2019-09-20 23:02:41.731] Begin Type declaration (phase 1) for 1 files
[2019-09-20 23:02:41.731] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 23:02:41.733] Type declaration (phase 1) for 1 files: 0.002258
[2019-09-20 23:02:41.734] Heap size: 4355712
[2019-09-20 23:02:41.734] Begin Type declaration (phase 2) for 1 files
[2019-09-20 23:02:41.734] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 23:02:41.734] Invalidating (but not recomputing) declarations in 1 files
[2019-09-20 23:02:41.737] Type declaration (phase 2) for 1 files: 0.002996
[2019-09-20 23:02:41.737] Heap size: 4356608
[2019-09-20 23:02:41.737] Begin evaluating prechecked changes
[2019-09-20 23:02:41.737] Evaluating prechecked changes: 0.000201
[2019-09-20 23:02:41.737] Begin Typechecking 1 files
[2019-09-20 23:02:41.737] Calculated whether should do remote type checking (false): 0.000004
[2019-09-20 23:02:41.737] Type checking service will process files sequentially
[2019-09-20 23:02:41.740] Typechecked 1 files: 0.003737
[2019-09-20 23:02:41.741] Total: 0.011525

[2019-09-20 23:02:41.741] Recheck id: kCIC9Nb8IM
[2019-09-20 23:02:41.741] Finished recheck_loop; sending push message
******************************************
[2019-09-20 23:02:42.487] Check kind: Full_check
[2019-09-20 23:02:42.487] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 23:02:42.487] Processing deferred type decl for 0 files
[2019-09-20 23:02:42.488] Processing deferred typechecking for 0 files
[2019-09-20 23:02:42.488] Begin Parsing 1 files
[2019-09-20 23:02:42.489] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 23:02:42.490] Updated search index for symbols in 1 files:: 0.000589
[2019-09-20 23:02:42.490] Done updating search index
[2019-09-20 23:02:42.490] Parsing 1 files: 0.002300
[2019-09-20 23:02:42.490] Heap size: 4357440
[2019-09-20 23:02:42.490] Begin Updating deps
[2019-09-20 23:02:42.490] Updating deps: 0.000085
[2019-09-20 23:02:42.490] Begin Naming
[2019-09-20 23:02:42.490] Naming: 0.000182
[2019-09-20 23:02:42.490] Begin Type declaration (phase 1) for 1 files
[2019-09-20 23:02:42.490] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 23:02:42.492] Type declaration (phase 1) for 1 files: 0.002357
[2019-09-20 23:02:42.492] Heap size: 4358528
[2019-09-20 23:02:42.493] Begin Type declaration (phase 2) for 1 files
[2019-09-20 23:02:42.493] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 23:02:42.493] Invalidating (but not recomputing) declarations in 0 files
[2019-09-20 23:02:42.495] Type declaration (phase 2) for 1 files: 0.002291
[2019-09-20 23:02:42.495] Heap size: 4359424
[2019-09-20 23:02:42.495] Begin evaluating prechecked changes
[2019-09-20 23:02:42.495] Evaluating prechecked changes: 0.000091
[2019-09-20 23:02:42.495] Begin Typechecking 1 files
[2019-09-20 23:02:42.495] Calculated whether should do remote type checking (false): 0.000005
[2019-09-20 23:02:42.495] Type checking service will process files sequentially
[2019-09-20 23:02:42.497] Typechecked 1 files: 0.002691
[2019-09-20 23:02:42.498] Total: 0.009997

[2019-09-20 23:02:42.498] Recheck id: xwNI2sWEjs
[2019-09-20 23:02:42.498] Finished recheck_loop; is_empty errors

```

**Comment by lexidor** [comment]
### Privacy notice

One or more things in this issue may be able to uniquely identify me and or my machine. I don't know if the `hh_server` log is very revealing. Can I kindly ask you to not use this to break my privacy? That would be very much appreciated.

**Comment by lexidor** [comment]
### Update

Not even a full shutdown resolved the issue. Here is the output from the attempt after the reboot.

```
hh_server --config symbolindex_search_provider=SqliteIndex .
[2019-09-20 23:20:45.128] Server version is not set
[2019-09-20 23:20:45.128] Loading config exception: (Sys_error "/etc/hh.conf: No such file or directory")
[2019-09-20 23:20:45.128] Could not load config at /etc/hh.conf
Config overrides:
symbolindex_search_provider = SqliteIndex

The combined config:
symbolindex_search_provider = SqliteIndex
Informant using dummy - resigning
[2019-09-20 23:20:45.134] Starting first server
[2019-09-20 23:20:45.134] About to spawn typechecker daemon. Logs will go here.
[2019-09-20 23:20:45.173] Just started typechecker server with pid: 2588.
[2019-09-20 23:20:45.241] Server version is not set
[2019-09-20 23:20:45.241] Loading config exception: (Sys_error "/etc/hh.conf: No such file or directory")
[2019-09-20 23:20:45.241] Could not load config at /etc/hh.conf
Config overrides:
symbolindex_search_provider = SqliteIndex

The combined config:
symbolindex_search_provider = SqliteIndex
[2019-09-20 23:20:45.242] Version: hackc-2b00aad2b355c924e392ce3d004894215f1ad9ba-4.24.0-dev
[2019-09-20 23:20:45.242] Hostname: lexidor
[2019-09-20 23:20:45.244] Using dfind
[2019-09-20 23:20:45.244] Initializing Server (This might take some time)
[2019-09-20 23:20:45.244] Init id: nQ3y1m3Ti8
[2019-09-20 23:20:45.244] Initing with approach: Local_config_saved_state_disabled
[2019-09-20 23:20:45.245] indexing
[2019-09-20 23:20:45.245] indexing: 0.000003
[2019-09-20 23:20:45.245] parsing
[2019-09-20 23:20:45.823] Heap size: 1587968
[2019-09-20 23:20:45.823] Parsing: 0.578389
[2019-09-20 23:20:45.823] Updating file dependencies...
[2019-09-20 23:20:45.826] Updated file dependencies: 0.002963
[2019-09-20 23:20:45.826] resolving symbol references
[2019-09-20 23:20:45.845] Heap size: 2448576
[2019-09-20 23:20:45.845] Naming: 0.019061
[2019-09-20 23:20:45.847] Unable to fetch sqlite symbol index: Saved states are not supported in this build.
[2019-09-20 23:20:45.847] Generating [/tmp/symbolindex.zSoptzSlamppzShtdocszSphq.1569014445.db] from repository [/opt/lampp/htdocs/phq/]
[2019-09-20 23:20:45.850] Scanned repository folder [/opt/lampp/htdocs/phq/] in  [0.0 secs]
[2019-09-20 23:20:45.853] Scanned HHI folder [/tmp/hh_server/hhi_31b51f67] in  [0.0 secs]
[2019-09-20 23:20:45.916] Parsed 206 files in  [0.1 secs]
[2019-09-20 23:20:45.925] Wrote 5133 symbols to sqlite in  [0.0 secs]
[2019-09-20 23:20:45.925] Initialized symbol index sqlite: [/tmp/symbolindex.zSoptzSlamppzShtdocszSphq.1569014445.db]
[2019-09-20 23:20:45.925] Search provider set to [Sqlite] based on configuration value [SqliteIndex]
[2019-09-20 23:20:45.925] Updated search index for symbols in 248 files:: 0.000029
[2019-09-20 23:20:45.925] Done updating search index
[2019-09-20 23:20:45.926] Dependency table load factor: 0 / 131072 (0.00)
[2019-09-20 23:20:45.926] Hashtable load factor: 8933 / 262144 (0.03) with 8933 nonempty slots
[2019-09-20 23:20:45.926] Waiting for daemon(s) to be ready...
[2019-09-20 23:20:45.926] Server is partially ready
[2019-09-20 23:20:45.926] Took 0.681358 seconds.
******************************************
[2019-09-20 23:20:46.027] Check kind: Full_check
[2019-09-20 23:20:46.027] Processing changes to 0 files
[2019-09-20 23:20:46.027] Processing deferred type decl for 0 files
[2019-09-20 23:20:46.027] Processing deferred typechecking for 248 files
[2019-09-20 23:20:46.027] Begin Parsing 0 files
[2019-09-20 23:20:46.027] Parsing 0 files: 0.000185
[2019-09-20 23:20:46.027] Heap size: 2448576
[2019-09-20 23:20:46.027] Begin Updating deps
[2019-09-20 23:20:46.027] Updating deps: 0.000081
[2019-09-20 23:20:46.027] Begin Naming
[2019-09-20 23:20:46.027] Naming: 0.000080
[2019-09-20 23:20:46.027] Begin Type declaration (phase 1) for 0 files
[2019-09-20 23:20:46.027] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 23:20:46.030] Type declaration (phase 1) for 0 files: 0.002487
[2019-09-20 23:20:46.030] Heap size: 2448576
[2019-09-20 23:20:46.030] Begin Type declaration (phase 2) for 0 files
[2019-09-20 23:20:46.030] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 23:20:46.030] Invalidating (but not recomputing) declarations in 0 files
[2019-09-20 23:20:46.032] Type declaration (phase 2) for 0 files: 0.001798
[2019-09-20 23:20:46.032] Heap size: 2448576
[2019-09-20 23:20:46.032] Begin evaluating prechecked changes
[2019-09-20 23:20:46.032] Evaluating prechecked changes: 0.000088
[2019-09-20 23:20:46.032] Extending the FAST by 248 additional files
[2019-09-20 23:20:46.032] Extending FAST sequentially (in memory)...
[2019-09-20 23:20:46.036] Extended FAST: 0.003888
[2019-09-20 23:20:46.036] Begin Typechecking 248 files
[2019-09-20 23:20:46.036] Calculated whether should do remote type checking (false): 0.000005
[2019-09-20 23:20:46.036] Type checking service will process files in parallel
[2019-09-20 23:20:46.315] Typechecked 248 files: 0.283494
[2019-09-20 23:20:46.315] Total: 0.288213

[2019-09-20 23:20:46.315] Dependency table load factor: 5438 / 131072 (0.04)
[2019-09-20 23:20:46.315] Hashtable load factor: 17503 / 262144 (0.07) with 17503 nonempty slots
[2019-09-20 23:20:46.315] Heap size: 4327872
[2019-09-20 23:20:46.315] Server is READY
[2019-09-20 23:20:46.315] Took 1.070925 seconds to initialize.
[2019-09-20 23:20:46.315] Recheck id: qzpYQSGbUO
[2019-09-20 23:20:46.315] Finished recheck_loop; no diag subscriptions
[2019-09-20 23:20:53.240] Got priority request for typechecker. Prior request 8.0 seconds ago
[2019-09-20 23:20:53.249] Diag_subscribe: SUBSCRIBE 0
[2019-09-20 23:20:53.250] Diag_subscribe: open, remains active
******************************************
[2019-09-20 23:20:57.664] Check kind: Lazy_check
[2019-09-20 23:20:57.664] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 23:20:57.664] Begin Parsing 1 files
[2019-09-20 23:20:57.664] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 23:20:57.665] Updated search index for symbols in 1 files:: 0.000310
[2019-09-20 23:20:57.665] Done updating search index
[2019-09-20 23:20:57.665] Parsing 1 files: 0.001074
[2019-09-20 23:20:57.665] Heap size: 4329216
[2019-09-20 23:20:57.665] Begin Updating deps
[2019-09-20 23:20:57.665] Updating deps: 0.000072
[2019-09-20 23:20:57.665] Begin Naming
[2019-09-20 23:20:57.665] Naming: 0.000103
[2019-09-20 23:20:57.665] Begin Type declaration (phase 1) for 1 files
[2019-09-20 23:20:57.665] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 23:20:57.666] Type declaration (phase 1) for 1 files: 0.000800
[2019-09-20 23:20:57.666] Heap size: 4330304
[2019-09-20 23:20:57.666] Begin Type declaration (phase 2) for 1 files
[2019-09-20 23:20:57.666] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 23:20:57.666] Invalidating (but not recomputing) declarations in 1 files
[2019-09-20 23:20:57.667] Type declaration (phase 2) for 1 files: 0.001022
[2019-09-20 23:20:57.667] Heap size: 4331200
[2019-09-20 23:20:57.667] Begin evaluating prechecked changes
[2019-09-20 23:20:57.667] Evaluating prechecked changes: 0.000053
[2019-09-20 23:20:57.667] Begin Typechecking 1 files
[2019-09-20 23:20:57.667] Calculated whether should do remote type checking (false): 0.000002
[2019-09-20 23:20:57.667] Type checking service will process files sequentially
[2019-09-20 23:20:57.668] Typechecked 1 files: 0.001310
[2019-09-20 23:20:57.668] Total: 0.004434

[2019-09-20 23:20:57.668] Recheck id: EXtGuTVGZA
[2019-09-20 23:20:57.669] Finished recheck_loop; sending push message
******************************************
[2019-09-20 23:20:58.409] Check kind: Lazy_check
[2019-09-20 23:20:58.410] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 23:20:58.410] Begin Parsing 1 files
[2019-09-20 23:20:58.410] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 23:20:58.410] Updated search index for symbols in 1 files:: 0.000271
[2019-09-20 23:20:58.410] Done updating search index
[2019-09-20 23:20:58.411] Parsing 1 files: 0.001157
[2019-09-20 23:20:58.411] Heap size: 4332800
[2019-09-20 23:20:58.411] Begin Updating deps
[2019-09-20 23:20:58.411] Updating deps: 0.000050
[2019-09-20 23:20:58.411] Begin Naming
[2019-09-20 23:20:58.411] Naming: 0.000123
[2019-09-20 23:20:58.411] Begin Type declaration (phase 1) for 1 files
[2019-09-20 23:20:58.411] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 23:20:58.412] Type declaration (phase 1) for 1 files: 0.000847
[2019-09-20 23:20:58.412] Heap size: 4333888
[2019-09-20 23:20:58.412] Begin Type declaration (phase 2) for 1 files
[2019-09-20 23:20:58.412] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 23:20:58.412] Invalidating (but not recomputing) declarations in 1 files
[2019-09-20 23:20:58.413] Type declaration (phase 2) for 1 files: 0.000900
[2019-09-20 23:20:58.413] Heap size: 4334784
[2019-09-20 23:20:58.413] Begin evaluating prechecked changes
[2019-09-20 23:20:58.413] Evaluating prechecked changes: 0.000055
[2019-09-20 23:20:58.413] Extending the FAST by 1 additional files
[2019-09-20 23:20:58.413] Extending FAST sequentially (in memory)...
[2019-09-20 23:20:58.413] Extended FAST: 0.000007
[2019-09-20 23:20:58.413] Begin Typechecking 1 files
[2019-09-20 23:20:58.413] Calculated whether should do remote type checking (false): 0.000003
[2019-09-20 23:20:58.413] Type checking service will process files sequentially
[2019-09-20 23:20:58.414] Typechecked 1 files: 0.001074
[2019-09-20 23:20:58.414] Total: 0.004206

[2019-09-20 23:20:58.414] Recheck id: imzvxsDHal
[2019-09-20 23:20:58.414] Finished recheck_loop; sending push message
******************************************
[2019-09-20 23:20:59.564] Check kind: Full_check
[2019-09-20 23:20:59.564] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 23:20:59.564] Processing deferred type decl for 0 files
[2019-09-20 23:20:59.564] Processing deferred typechecking for 0 files
[2019-09-20 23:20:59.564] Begin Parsing 1 files
[2019-09-20 23:20:59.565] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 23:20:59.565] Updated search index for symbols in 1 files:: 0.000279
[2019-09-20 23:20:59.565] Done updating search index
[2019-09-20 23:20:59.565] Parsing 1 files: 0.000976
[2019-09-20 23:20:59.565] Heap size: 4335872
[2019-09-20 23:20:59.565] Begin Updating deps
[2019-09-20 23:20:59.565] Updating deps: 0.000047
[2019-09-20 23:20:59.565] Begin Naming
[2019-09-20 23:20:59.565] Naming: 0.000101
[2019-09-20 23:20:59.565] Begin Type declaration (phase 1) for 1 files
[2019-09-20 23:20:59.565] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 23:20:59.566] Type declaration (phase 1) for 1 files: 0.000960
[2019-09-20 23:20:59.566] Heap size: 4336960
[2019-09-20 23:20:59.566] Begin Type declaration (phase 2) for 1 files
[2019-09-20 23:20:59.566] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 23:20:59.566] Invalidating (but not recomputing) declarations in 0 files
[2019-09-20 23:20:59.567] Type declaration (phase 2) for 1 files: 0.000830
[2019-09-20 23:20:59.567] Heap size: 4337856
[2019-09-20 23:20:59.567] Begin evaluating prechecked changes
[2019-09-20 23:20:59.567] Evaluating prechecked changes: 0.000049
[2019-09-20 23:20:59.567] Extending the FAST by 1 additional files
[2019-09-20 23:20:59.567] Extending FAST sequentially (in memory)...
[2019-09-20 23:20:59.567] Extended FAST: 0.000008
[2019-09-20 23:20:59.567] Begin Typechecking 1 files
[2019-09-20 23:20:59.567] Calculated whether should do remote type checking (false): 0.000019
[2019-09-20 23:20:59.567] Type checking service will process files sequentially
[2019-09-20 23:20:59.569] Typechecked 1 files: 0.001334
[2019-09-20 23:20:59.569] Total: 0.004297

[2019-09-20 23:20:59.569] Recheck id: EfSEBzpXy4
[2019-09-20 23:20:59.569] Finished recheck_loop; sending push message
******************************************
[2019-09-20 23:20:59.993] Check kind: Lazy_check
[2019-09-20 23:20:59.993] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 23:20:59.993] Begin Parsing 1 files
[2019-09-20 23:20:59.994] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 23:20:59.994] Updated search index for symbols in 1 files:: 0.000284
[2019-09-20 23:20:59.994] Done updating search index
[2019-09-20 23:20:59.994] Parsing 1 files: 0.001120
[2019-09-20 23:20:59.994] Heap size: 4338944
[2019-09-20 23:20:59.994] Begin Updating deps
[2019-09-20 23:20:59.994] Updating deps: 0.000051
[2019-09-20 23:20:59.994] Begin Naming
[2019-09-20 23:20:59.994] Naming: 0.000120
[2019-09-20 23:20:59.995] Begin Type declaration (phase 1) for 1 files
[2019-09-20 23:20:59.995] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 23:20:59.995] Type declaration (phase 1) for 1 files: 0.000871
[2019-09-20 23:20:59.995] Heap size: 4340032
[2019-09-20 23:20:59.995] Begin Type declaration (phase 2) for 1 files
[2019-09-20 23:20:59.995] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 23:20:59.995] Invalidating (but not recomputing) declarations in 1 files
[2019-09-20 23:20:59.996] Type declaration (phase 2) for 1 files: 0.000963
[2019-09-20 23:20:59.996] Heap size: 4340928
[2019-09-20 23:20:59.996] Begin evaluating prechecked changes
[2019-09-20 23:20:59.996] Evaluating prechecked changes: 0.000052
[2019-09-20 23:20:59.996] Extending the FAST by 1 additional files
[2019-09-20 23:20:59.996] Extending FAST sequentially (in memory)...
[2019-09-20 23:20:59.996] Extended FAST: 0.000007
[2019-09-20 23:20:59.996] Begin Typechecking 1 files
[2019-09-20 23:20:59.997] Calculated whether should do remote type checking (false): 0.000024
[2019-09-20 23:20:59.997] Type checking service will process files sequentially
[2019-09-20 23:20:59.998] Typechecked 1 files: 0.001270
[2019-09-20 23:20:59.998] Total: 0.004447

[2019-09-20 23:20:59.998] Recheck id: qan3WwVPzT
[2019-09-20 23:20:59.998] Finished recheck_loop; sending push message
******************************************
[2019-09-20 23:21:00.794] Check kind: Full_check
[2019-09-20 23:21:00.794] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 23:21:00.794] Processing deferred type decl for 0 files
[2019-09-20 23:21:00.794] Processing deferred typechecking for 0 files
[2019-09-20 23:21:00.794] Begin Parsing 1 files
[2019-09-20 23:21:00.794] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 23:21:00.795] Updated search index for symbols in 1 files:: 0.000289
[2019-09-20 23:21:00.795] Done updating search index
[2019-09-20 23:21:00.795] Parsing 1 files: 0.000950
[2019-09-20 23:21:00.795] Heap size: 4341760
[2019-09-20 23:21:00.795] Begin Updating deps
[2019-09-20 23:21:00.795] Updating deps: 0.000055
[2019-09-20 23:21:00.795] Begin Naming
[2019-09-20 23:21:00.795] Naming: 0.000172
[2019-09-20 23:21:00.795] Begin Type declaration (phase 1) for 1 files
[2019-09-20 23:21:00.795] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 23:21:00.796] Type declaration (phase 1) for 1 files: 0.001023
[2019-09-20 23:21:00.796] Heap size: 4342848
[2019-09-20 23:21:00.796] Begin Type declaration (phase 2) for 1 files
[2019-09-20 23:21:00.796] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 23:21:00.796] Invalidating (but not recomputing) declarations in 0 files
[2019-09-20 23:21:00.797] Type declaration (phase 2) for 1 files: 0.000855
[2019-09-20 23:21:00.797] Heap size: 4343744
[2019-09-20 23:21:00.797] Begin evaluating prechecked changes
[2019-09-20 23:21:00.797] Evaluating prechecked changes: 0.000050
[2019-09-20 23:21:00.797] Extending the FAST by 1 additional files
[2019-09-20 23:21:00.797] Extending FAST sequentially (in memory)...
[2019-09-20 23:21:00.797] Extended FAST: 0.000025
[2019-09-20 23:21:00.797] Begin Typechecking 1 files
[2019-09-20 23:21:00.797] Calculated whether should do remote type checking (false): 0.000002
[2019-09-20 23:21:00.797] Type checking service will process files sequentially
[2019-09-20 23:21:00.798] Typechecked 1 files: 0.001238
[2019-09-20 23:21:00.798] Total: 0.004343

[2019-09-20 23:21:00.798] Recheck id: RJRc4gNFUE
[2019-09-20 23:21:00.798] Finished recheck_loop; is_empty errors
******************************************
[2019-09-20 23:21:01.969] Check kind: Lazy_check
[2019-09-20 23:21:01.969] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 23:21:01.969] Begin Parsing 1 files
[2019-09-20 23:21:01.969] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 23:21:01.970] Updated search index for symbols in 1 files:: 0.000246
[2019-09-20 23:21:01.970] Done updating search index
[2019-09-20 23:21:01.970] Parsing 1 files: 0.001103
[2019-09-20 23:21:01.970] Heap size: 4344832
[2019-09-20 23:21:01.970] Begin Updating deps
[2019-09-20 23:21:01.970] Updating deps: 0.000053
[2019-09-20 23:21:01.970] Begin Naming
[2019-09-20 23:21:01.970] Naming: 0.000099
[2019-09-20 23:21:01.970] Begin Type declaration (phase 1) for 1 files
[2019-09-20 23:21:01.970] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 23:21:01.971] Type declaration (phase 1) for 1 files: 0.000852
[2019-09-20 23:21:01.971] Heap size: 4345920
[2019-09-20 23:21:01.971] Begin Type declaration (phase 2) for 1 files
[2019-09-20 23:21:01.971] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 23:21:01.971] Invalidating (but not recomputing) declarations in 1 files
[2019-09-20 23:21:01.972] Type declaration (phase 2) for 1 files: 0.000801
[2019-09-20 23:21:01.972] Heap size: 4346816
[2019-09-20 23:21:01.972] Begin evaluating prechecked changes
[2019-09-20 23:21:01.972] Evaluating prechecked changes: 0.000053
[2019-09-20 23:21:01.972] Extending the FAST by 1 additional files
[2019-09-20 23:21:01.972] Extending FAST sequentially (in memory)...
[2019-09-20 23:21:01.972] Extended FAST: 0.000009
[2019-09-20 23:21:01.972] Begin Typechecking 1 files
[2019-09-20 23:21:01.972] Calculated whether should do remote type checking (false): 0.000002
[2019-09-20 23:21:01.972] Type checking service will process files sequentially
[2019-09-20 23:21:01.973] Typechecked 1 files: 0.001077
[2019-09-20 23:21:01.973] Total: 0.004038

[2019-09-20 23:21:01.973] Recheck id: dBz5ccHnK
[2019-09-20 23:21:01.973] Finished recheck_loop; sending push message
******************************************
[2019-09-20 23:21:02.969] Check kind: Full_check
[2019-09-20 23:21:02.969] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 23:21:02.969] Processing deferred type decl for 0 files
[2019-09-20 23:21:02.969] Processing deferred typechecking for 0 files
[2019-09-20 23:21:02.969] Begin Parsing 1 files
[2019-09-20 23:21:02.970] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 23:21:02.970] Updated search index for symbols in 1 files:: 0.000280
[2019-09-20 23:21:02.970] Done updating search index
[2019-09-20 23:21:02.970] Parsing 1 files: 0.000869
[2019-09-20 23:21:02.970] Heap size: 4347904
[2019-09-20 23:21:02.970] Begin Updating deps
[2019-09-20 23:21:02.970] Updating deps: 0.000054
[2019-09-20 23:21:02.970] Begin Naming
[2019-09-20 23:21:02.970] Naming: 0.000164
[2019-09-20 23:21:02.970] Begin Type declaration (phase 1) for 1 files
[2019-09-20 23:21:02.970] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 23:21:02.971] Type declaration (phase 1) for 1 files: 0.000885
[2019-09-20 23:21:02.971] Heap size: 4348992
[2019-09-20 23:21:02.971] Begin Type declaration (phase 2) for 1 files
[2019-09-20 23:21:02.971] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 23:21:02.971] Invalidating (but not recomputing) declarations in 0 files
[2019-09-20 23:21:02.972] Type declaration (phase 2) for 1 files: 0.000868
[2019-09-20 23:21:02.972] Heap size: 4349888
[2019-09-20 23:21:02.972] Begin evaluating prechecked changes
[2019-09-20 23:21:02.972] Evaluating prechecked changes: 0.000051
[2019-09-20 23:21:02.972] Extending the FAST by 1 additional files
[2019-09-20 23:21:02.972] Extending FAST sequentially (in memory)...
[2019-09-20 23:21:02.972] Extended FAST: 0.000008
[2019-09-20 23:21:02.972] Begin Typechecking 1 files
[2019-09-20 23:21:02.972] Calculated whether should do remote type checking (false): 0.000024
[2019-09-20 23:21:02.972] Type checking service will process files sequentially
[2019-09-20 23:21:02.974] Typechecked 1 files: 0.001267
[2019-09-20 23:21:02.974] Total: 0.004158

[2019-09-20 23:21:02.974] Recheck id: SsirCquD4W
[2019-09-20 23:21:02.974] Finished recheck_loop; sending push message
******************************************
[2019-09-20 23:21:04.092] Check kind: Full_check
[2019-09-20 23:21:04.092] Processing changes to 1 file: /opt/lampp/htdocs/phq/src/code/Actor/User.hack
[2019-09-20 23:21:04.092] Processing deferred type decl for 0 files
[2019-09-20 23:21:04.092] Processing deferred typechecking for 0 files
[2019-09-20 23:21:04.092] Begin Parsing 1 files
[2019-09-20 23:21:04.093] src/code/Actor/User.hack - classes: \Lex\Actor\User
[2019-09-20 23:21:04.094] Updated search index for symbols in 1 files:: 0.000473
[2019-09-20 23:21:04.094] Done updating search index
[2019-09-20 23:21:04.094] Parsing 1 files: 0.001819
[2019-09-20 23:21:04.094] Heap size: 4350720
[2019-09-20 23:21:04.094] Begin Updating deps
[2019-09-20 23:21:04.094] Updating deps: 0.000075
[2019-09-20 23:21:04.094] Begin Naming
[2019-09-20 23:21:04.094] Naming: 0.000160
[2019-09-20 23:21:04.094] Begin Type declaration (phase 1) for 1 files
[2019-09-20 23:21:04.094] (Recomputing type declarations in changed files and determining immediate typechecking fanout)
[2019-09-20 23:21:04.096] Type declaration (phase 1) for 1 files: 0.002014
[2019-09-20 23:21:04.096] Heap size: 4351808
[2019-09-20 23:21:04.096] Begin Type declaration (phase 2) for 1 files
[2019-09-20 23:21:04.096] (Recomputing type declarations for descendants of changed classes and determining full typechecking fanout)
[2019-09-20 23:21:04.096] Invalidating (but not recomputing) declarations in 0 files
[2019-09-20 23:21:04.098] Type declaration (phase 2) for 1 files: 0.001934
[2019-09-20 23:21:04.098] Heap size: 4352704
[2019-09-20 23:21:04.098] Begin evaluating prechecked changes
[2019-09-20 23:21:04.098] Evaluating prechecked changes: 0.000079
[2019-09-20 23:21:04.098] Begin Typechecking 1 files
[2019-09-20 23:21:04.098] Calculated whether should do remote type checking (false): 0.000004
[2019-09-20 23:21:04.098] Type checking service will process files sequentially
[2019-09-20 23:21:04.101] Typechecked 1 files: 0.002575
[2019-09-20 23:21:04.101] Total: 0.008656

[2019-09-20 23:21:04.101] Recheck id: ViwsKuFzCj
[2019-09-20 23:21:04.101] Finished recheck_loop; is_empty errors
```

**Comment by lexidor** [comment]
### For reproducability

Without my code, just the HSL.

```HACK
function foo(HH\Lib\Re<autocomplete>): void {}
```
Expect `Ref`.

Funny sidenote, autocomplete will help you with autocompleting name namespace components, `HH\L` autocompletes to `HH\Lib`. But `HH\Lib\Re` only autocompletes to `HH\Lib\Regex`.

**Comment by lexidor** [comment]
Fixed
